x-pack/test/api_integration/apis/ml/filters/update_filters.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common';
+
+// eslint-disable-next-line import/no-default-export
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const items = ['104.236.210.185'];
+  const validFilters = [
+    {
+      filterId: 'filter_power',
+      requestBody: { description: 'Test delete filter #1', items },
+    },
+    {
+      filterId: 'filter_viewer',
+      requestBody: { description: 'Test delete filter (viewer)', items },
+    },
+    {
+      filterId: 'filter_unauthorized',
+      requestBody: { description: 'Test delete filter (unauthorized)', items },
+    },
+  ];
+
+  describe('update_filters', function () {
+    const updateFilterRequest = {
+      description: 'Updated filter #1',
+      removeItems: items,
+      addItems: ['my_new_items_1', 'my_new_items_2'],
+    };
+    before(async () => {
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      for (const filter of validFilters) {
+        const { filterId, requestBody } = filter;
+        await ml.api.createFilter(filterId, requestBody);
+      }
+    });
+
+    after(async () => {
+      for (const filter of validFilters) {
+        const { filterId } = filter;
+        await ml.api.deleteFilter(filterId);
+      }
+    });
+
+    it(`should update filter by id`, async () => {
+      const { filterId } = validFilters[0];
+      const { body } = await supertest
+        .put(`/api/ml/filters/${filterId}`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(updateFilterRequest)
+        .expect(200);
+
+      expect(body.filter_id).to.eql(filterId);
+      expect(body.description).to.eql(updateFilterRequest.description);
+      expect(body.items).to.eql(updateFilterRequest.addItems);
+    });
+
+    it(`should not allow to update filter for user without required permission`, async () => {
+      const { filterId } = validFilters[1];
+      const { body } = await supertest
+        .put(`/api/ml/filters/${filterId}`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(updateFilterRequest)
+        .expect(404);
+      expect(body.error).to.eql('Not Found');
```

Comment:
Would be good to also validate that the filter hasn't changed (similar to the test case below).

Reply:
Added in https://github.com/elastic/kibana/pull/72564/commits/121d1e5fdd860b91516a4ae3e2023cec45205c9b